### PR TITLE
update LICENSE.md for third-party sources

### LINKED ISSUE #520: Legality of include config.guess in the source
The GNU config.guess script is licensed under GPLv3 with a special permission:

```sh
# As a special exception to the GNU General Public License, if you
# distribute this file as part of a program that contains a
# configuration script generated by Autoconf, you may include it under
# the same distribution terms that you use for the rest of that
# program.  This Exception is an additional permission under section 7
# of the GNU General Public License, version 3 ("GPLv3").
```

MXE is not using GNU Autoconf in any way, so is it legal to distribute it under the MIT license?

### PR DESCRIPTION
**Pull request by tonytheodore**
Following on from #1503, the actual text probably needs updating.

Largely inspired by https://github.com/JuliaLang/julia/blob/master/LICENSE.md

Closes #520

### CODE REVIEW

File: LICENSE.md [hunk 1]
```diff
@@ -1,33 +1,49 @@
...
-  * Tony Theodore
-  * Martin Gerhardy
-  * Tiancheng "Timothy" Gu
-  * Boris Nagaev
-  * ... and many other contributors
+MXE distributes code from the following projects in the `ext/` directory:
 
-(contact via the project mailing list)
+- [GNU config.guess][config.guess] [GPL3+ with Autoconf exception]
+- [GNU Make Standard Library][gmsl] [BSD 3-clause]
```
Comment (starius) [inline LICENSE.md]: Could you provide links to texts of "GPL3+ with Autoconf exception" and "BSD 3-clause", please?

File: LICENSE.md [hunk 2]
```diff
@@ -1,33 +1,49 @@
...
-SOFTWARE.
+The [packages][packages] downloaded by MXE (ending up in `pkg/`) each
+have their own licence(s). Please see the individual source trees and/or
+websites for further details.
 
+[contributors]:https://github.com/mxe/mxe/graphs/contributors
+[mailing-list]:https://lists.nongnu.org/mailman/listinfo/mingw-cross-env-list
+[issue-tracker]:https://github.com/mxe/mxe/issues
+[config.guess]:http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD
+[gmsl]:https://sourceforge.net/p/gmsl/gmsl/ci/master/tree/__gmsl
```
Comment (starius) [inline LICENSE.md]: Could you provide links to revisions of config.guess and gmsl in use by MXE, please?

File: LICENSE.md [hunk 3]
```diff
@@ -1,33 +1,49 @@
...
+[packages]:http://mxe.cc/#packages
```
Comment (starius) [inline LICENSE.md]: Could you add space ` ` after `]:`, please?

**Comment by tonytheodore** [comment]
Links fixed by pointing to the MXE repo and highlighting licence section.

Preview: https://github.com/tonytheodore/mxe/blob/licence/LICENSE.md

**Comment by starius** [comment]
Looks good to me!

**Pull request merged**